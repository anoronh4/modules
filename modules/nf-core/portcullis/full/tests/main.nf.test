nextflow_process {

    name "Test Process PORTCULLIS_FULL"
    script "../main.nf"
    process "PORTCULLIS_FULL"

    tag "modules"
    tag "modules_nfcore"
    tag "portcullis"
    tag "portcullis/full"

    test("portcullis_full_paired_end") {
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.rna.paired_end.bam', checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/tab/test_rnaseq_SJ.out.tab', checkIfExists: true)
                    ]
                input[2] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.pass_junctions_bed,
                    process.out.pass_junctions_tab,
                    process.out.intron_gff,
                    process.out.exon_gff,
                    process.out.spliced_bam,
                    process.out.spliced_bai,
                    process.out.versions
                    ).match()
                }
            )
        }
    }

    test("portcullis_full_paired_end_stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.rna.paired_end.bam', checkIfExists: true)
                    ]
                input[1] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/tab/test_rnaseq_SJ.out.tab', checkIfExists: true)
                    ]
                input[2] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.pass_junctions_bed,
                    process.out.pass_junctions_tab,
                    process.out.intron_gff,
                    process.out.exon_gff,
                    process.out.versions
                    ).match()
                }
            )
        }
    }

}
