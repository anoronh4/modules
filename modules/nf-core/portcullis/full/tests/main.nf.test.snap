{
    "portcullis_full_paired_end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "portcullis_filtered.pass.junctions.bed:md5,613288f6b446359fd0153a7e628e1489"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "portcullis_filtered.pass.junctions.tab:md5,6e2dd7e93f6b7a23d90d3c3c11e99085"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "portcullis_filtered.pass.junctions.intron.gff3:md5,3ee2c3526bfb54f81bfd55a3ab679570"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "portcullis_filtered.pass.junctions.exon.gff3:md5,7a3c56aca8dcd39ee2dec40589511a81"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "portcullis_all.spliced.bam:md5,3357d8e89ba799bce5a8c76f10e247e6"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "portcullis_all.spliced.bam.bai:md5,03e4b328e0200d24d49e2316d3d78f4c"
                ]
            ],
            [
                "versions.yml:md5,b297b098450e5baaea436e822fe25b0e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-31T16:48:18.276728"
    },
    "portcullis_full_paired_end_stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.pass.junctions.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.pass.junctions.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                "versions.yml:md5,b297b098450e5baaea436e822fe25b0e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-31T16:48:26.392204"
    }
}